java/ql/src/semmle/code/java/frameworks/guava/IO.qll

```diff
@@ -0,0 +1,66 @@
+        "com.google.common.io;ByteSource;true;openStream;();;Argument[-1];ReturnValue;taint",
+        "com.google.common.io;ByteSource;true;openBufferedStream;();;Argument[-1];ReturnValue;taint",
+        "com.google.common.io;ByteSource;true;read;();;Argument[-1];ReturnValue;taint",
+        "com.google.common.io;ByteSource;true;slice;(long,long);;Argument[-1];ReturnValue;taint",
+        "com.google.common.io;ByteSource;true;wrap;(byte[]);;Argument[0];ReturnValue;taint",
+        "com.google.common.io;ByteStreams;false;copy;(InputStream,OutputStream);;Argument[0];Argument[1];taint",
+        "com.google.common.io;ByteStreams;false;copy;(ReadablyByteChannel,WritableByteChannel);;Argument[0];Argument[1];taint",
+        "com.google.common.io;ByteStreams;false;limit;(InputStream,long);;Argument[0];ReturnValue;taint",
+        "com.google.common.io;ByteStreams;false;newDataInput;(byte[]);;Argument[0];ReturnValue;taint",
+        "com.google.common.io;ByteStreams;false;newDataInput;(byte[],int);;Argument[0];ReturnValue;taint",
+        "com.google.common.io;ByteStreams;false;newDataInput;(ByteArrayInputStream);;Argument[0];ReturnValue;taint",
+        "com.google.common.io;ByteStreams;false;read;(InputStream,byte[],int,int);;Argument[0];Argument[1];taint",
+        "com.google.common.io;ByteStreams;false;readFully;(InputStream,byte[]);;Argument[0];Argument[1];taint",
+        "com.google.common.io;ByteStreams;false;readFully;(InputStream,byte[],int,int);;Argument[0];Argument[1];taint",
+        "com.google.common.io;ByteStreams;false;toByteArray;(InputStream);;Argument[0];ReturnValue;taint",
+        "com.google.common.io;CharSource;true;asByteSource;(Charset);;Argument[-1];ReturnValue;taint",
+        "com.google.common.io;CharSource;true;concat;;;Argument[0];ReturnValue;taint",
+        "com.google.common.io;CharSource;true;copyTo;(Appendable);;Argument[-1];Argument[0];taint",
+        "com.google.common.io;CharSource;true;openStream;();;Argument[-1];ReturnValue;taint",
+        "com.google.common.io;CharSource;true;openBufferedStream;();;Argument[-1];ReturnValue;taint",
+        "com.google.common.io;CharSource;true;read;();;Argument[-1];ReturnValue;taint",
+        "com.google.common.io;CharSource;true;readFirstLine;();;Argument[-1];ReturnValue;taint",
+        "com.google.common.io;CharSource;true;readLines;();;Argument[-1];ReturnValue;taint",
+        "com.google.common.io;CharSource;true;lines;();;Argument[-1];ReturnValue;taint",
+        "com.google.common.io;CharSource;true;wrap;(CharSequence);;Argument[0];ReturnValue;taint",
+        "com.google.common.io;CharStreams;false;copy;(Readable,Appendable);;Argument[0];Argument[1];taint",
+        "com.google.common.io;CharStreams;false;readLines;(Readable);;Argument[0];ReturnValue;taint",
+        "com.google.common.io;CharStreams;false;toString;(Readable);;Argument[0];ReturnValue;taint",
+        "com.google.common.io;Closer;true;register;;;Argument[0];ReturnValue;value",
```

Comment:
Should exception objects also be tracked for taint? E.g. when an exception message is created with a user-defined value and is then logged, it could allow log injection. Though in general any flow from user-defined data to an exception constructor is bad, so maybe it is not worth trying to check where the tainted exception flows afterwards?